### PR TITLE
Fixes in exportable code

### PR DESCRIPTION
**Pull request by GalyaZalesskaya**
#### Fixes in exportable code:

- Fixed torch imports in `otx/algorithms/common/utils/utils.py`, so the demo in exportable code doesn't require torch to run anymore
- Removed the condition that ACTION_CLASSIFICATION and ACTION_DETECTION TaskType and Domain appear in enum only if `FEATURE_FLAGS_OTX_ACTION_TASKS` flag is used. This caused an issue that [enum initialization](https://github.com/openvinotoolkit/training_extensions/blob/6cbac4689979fc44df20532c7207497af38bf072/otx/api/entities/model_template.py#L117) happened before initializing FEATURE_FLAGS_OTX_ACTION_TASKS as 1 for CLI tasks [here](https://github.com/openvinotoolkit/training_extensions/blob/63a1a0408c1e0310d3eb31b21aaebb13ff4ae0b7/otx/cli/__init__.py#L11), so the demo failed with attribute error for ACTION CLASSIFICATION.

#### Enhances:
- Removed coping `__pycache__` to deployed zip from exportable code  
- Moved utility function for saving the demo output to `otx.api.utils` to avoid cross-dependencies on otx.cli in exportable code package

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [x] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: otx/api/entities/label.py [hunk 1]
```diff
@@ -26,9 +26,8 @@ class Domain(Enum):
     ANOMALY_SEGMENTATION = auto()
     INSTANCE_SEGMENTATION = auto()
     ROTATED_DETECTION = auto()
-    if os.getenv("FEATURE_FLAGS_OTX_ACTION_TASKS", "0") == "1":
-        ACTION_CLASSIFICATION = auto()
-        ACTION_DETECTION = auto()
+    ACTION_CLASSIFICATION = auto()
```
Comment (sungmanc) [inline otx/api/entities/label.py]: As far as I know, this flag is important for Geti. could you check? @goodsong81

File: otx/api/entities/model_template.py [hunk 2]
```diff
@@ -205,17 +205,16 @@ def __new__(cls, *args):
...
-            domain=Domain.ACTION_CLASSIFICATION,
-            is_trainable=True,
-            is_anomaly=False,
-            is_global=False,
-            is_local=True,
-        )
-        ACTION_DETECTION = 15, TaskInfo(
-            domain=Domain.ACTION_DETECTION, is_trainable=True, is_anomaly=False, is_global=False, is_local=True
-        )
+    ACTION_CLASSIFICATION = 14, TaskInfo(
```
Comment (sungmanc) [inline otx/api/entities/model_template.py]: Same with above

**Review by sungmanc — COMMENTED** [review]
Left some comments.

**Review by goodsong81 — CHANGES_REQUESTED** [review]
ACTION_CLASSIFICATION/DETECTION should not be exposed to Geti yet, it's the reason why we adopted the `FEATURE_FLAG` mechanism. Can you fix the issue in other ways?

If you mean the exportable code `demo.py`, why don't you add `os.environ["FEATURE_FLAGS_OTX_ACTION_TASKS"] = "1"` in the `demo.py` before importing `otx`?

File: otx/algorithms/action/adapters/openvino/task.py [hunk 3]
```diff
@@ -247,6 +247,9 @@ def deploy(self, output_model: ModelEntity) -> None:
             arch.writestr(os.path.join("model", "config.json"), json.dumps(parameters, ensure_ascii=False, indent=4))
             # model_wrappers files
             for root, _, files in os.walk(os.path.dirname(model_wrappers.__file__)):
+                # skip coping __pycache__ folder
```
Comment (sovrasov) [inline otx/algorithms/action/adapters/openvino/task.py]: It looks like this comment is useless

Comment (GalyaZalesskaya) [reply otx/algorithms/action/adapters/openvino/task.py]: Removed them, thank you

**Comment by GalyaZalesskaya** [comment]
@sungmanc @goodsong81 Thank you for the clarification about ACTION_CLASSIFICATION/DETECTION flag, I fixed its usage

**Review by goodsong81 — APPROVED** [review]

**Pull request merged**